### DISCUSSION TITLE
BNF available for mojo?

### DISCUSSION THREAD

**Discussion by chwmilcon** [modular/modular#401] [General]
With it understood that Mojo is a work in progress, is there a BNF available for the language as it is today? Or how it is imagined to be in the future? Pardon me if there's one already out there and I missed it.

**Comment by lattner** [comment]
We currently use a hand written parser in C++ and don't have a formal grammar unfortunately; we do have a black grammar, but it is "quite a superset" of the language.  It would be great to build this out and document it.

**Comment by llllvvuu** [comment]
This would be very useful for [tree-sitter](https://tree-sitter.github.io)-based applications.

**Comment by StandinKP** [comment]
Are we working on building the grammar for this? Would love to help out with it if that is in the bucket
